pkgs/stdenv/darwin/default.nix

```diff
@@ -1,10 +1,34 @@
+# This file contains the standard build environment for Darwin. It is based on LLVM and is patterned
+# after the Linux stdenv. It shares similar goals to the Linux standard environment in that the
+# resulting environment should be built purely and not contain any references to it.
+#
+# There are two more goals worth calling out explicitly:
+# 1. The standard environment should build successfully with sandboxing enabled on Darwin. It is
+#    fine if a package requires a `sandboxProfile` to build, but it should not be necessary to
+#    disable the sandbox to build the stdenv successfully; and
+# 2. The output should depend weakly on the bootstrap tools. Historically, Darwin required updating
+#    the bootstrap tools prior to updating the version of LLVM used in the standard environment.
+#    By not depending on a specific version, the LLVM used on Darwin can be updated simply by
+#    bumping the definition of llvmPackages in `all-packages.nix`.
+#
+# There are effectively two steps when updating the standard environment:
+# 1. Update the definition of llvmPackages in `all-packages.nix` for Darwin to match the value of
+#    llvmPackages.latest in `all-packages.nix`. Timing-wise, this done currently using the spring
+#    release of LLVM and once llvmPackages.latest has been updated to match. If the LLVM project
+#    has announced a release schedule of patch updates, wait until those are in nixpkgs. Otherwise,
+#    the LLVM updates will have to go through staging instead of being merged into master; and
+# 2. Fix the resulting breakage. Most things break due to additional warnings being turned into
+#    errors or additional strictness applied by LLVM. Fixes may come in the form of disabling those
+#    new warnings or by fixing the actual source (e.g., with a patch or update upstream). If the
+#    fix is trivial (e.g., adding a missing int to an implicit declaration), it is better to fix
+#    the problem instead of silencing the warning.
+
```

Comment:
The Linux stdenv starts with a lengthy comment explaining the stdenv’s goals and how to work on it. I’d like to follow that style and precedent. I’d also worry that people working on the stdenv could miss the README (unless they were pointed to it).

Reply:
I'd be in favour of a "proper" doc in that case too ;)

Pointing to it is certainly necessary.

Reply:
I’ve move it out and also made reference to the Linux stdenv because it also has some good documentation. I’m going to mark this as resolved, but feel free to unresolve if you’d like to see additional changes.